### PR TITLE
Added initContainer in server deployment to wait for MongoDB.

### PR DESCRIPTION
**Pull request by Jonsy13**
Signed-off-by: Jonsy13 <vedant.shrotria@chaosnative.com>

## Proposed changes

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

**This PR will ad initContainer in server deployment to wait for MongoDB**

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [x] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/cluster-k8s-manifest.yml [hunk 1]
```diff
@@ -530,6 +530,14 @@ spec:
       labels:
         component: litmusportal-server
     spec:
+      initContainers:
+        - name: wait-for-mongodb
+          image: curlimages/curl:latest
```
Comment (rajdas98) [inline litmus-portal/cluster-k8s-manifest.yml:535]: can you check alpine curl image?

Comment (Jonsy13) [reply litmus-portal/cluster-k8s-manifest.yml:535]: Sure

File: litmus-portal/cluster-k8s-manifest.yml [hunk 2]
```diff
@@ -530,6 +530,14 @@ spec:
...
+          command: ["/bin/sh", "-c"]
+          args:
+            [
+              "while [[ $(curl -sw '%{http_code}' http://mongo-service.$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace).svc.cluster.local:27017 -o /dev/null) -ne 200 ]]; do sleep 5; echo 'Waiting for the MongoDB to be ready...'; done; echo 'Connection with MongoDB established'",
```
Comment (gdsoumya) [inline litmus-portal/cluster-k8s-manifest.yml:539]: ns name should not be needed I guess, you can directly just use `mongo-service` like we do in the server

**Review by gdsoumya — CHANGES_REQUESTED** [review]

**Review by ishangupta-ds — APPROVED** [review]

File: litmus-portal/cluster-k8s-manifest.yml [hunk 3]
```diff
@@ -530,6 +530,14 @@ spec:
       labels:
         component: litmusportal-server
     spec:
+      initContainers:
+        - name: wait-for-mongodb
+          image: curlimages/curl:7.78.0
```
Comment (gdsoumya) [inline litmus-portal/cluster-k8s-manifest.yml:535]: We need to probably put this image in litmuschaos docker repo, but that can be done before the release

**Review by gdsoumya — APPROVED** [review]

File: litmus-portal/cluster-k8s-manifest.yml (hunk 3 above)
Comment (Jonsy13) [reply litmus-portal/cluster-k8s-manifest.yml:535]: Yes, will have to do that before release.

File: litmus-portal/cluster-k8s-manifest.yml (hunk 1 above)
Comment (Jonsy13) [reply litmus-portal/cluster-k8s-manifest.yml]: @rajdas98 Currently alpine image containing curl are not multi-arch, This is only multi-arch. We will add one image in litmuschaos repo for curl with multi-arch support.

**Comment by rajdas98** [comment]
Please add this to litmus-helm

**Pull request merged**